src/accel/TrackingManagerConstructor.hh

```diff
@@ -50,11 +50,18 @@ class TrackingManagerConstructor final : public G4VPhysicsConstructor
     //!@{
     //! \name Type aliases
     using LocalTransporterFromThread = std::function<LocalTransporter*(int)>;
+    using VecG4PD = std::vector<G4ParticleDefinition*>;
```

Comment:
Since the ordering for these doesn't matter (I just used a `Span` originally because the data was easily stored as a static array) let's use
```suggestion
    using SetG4PD = std::unordered_set<G4ParticleDefinition*>;
```
and return const references to static SetG4PD. That should also simplify your testing for user-requested particles.